scripts/update-haskell-backend.sh

```diff
@@ -12,12 +12,4 @@ if [ $REV != $OLD_REV ]; then
   sed -i -r "/haskell-backend.git/{n;s/tag:.*$/tag: $REV/}" cabal.project
   # Update the stack.yaml.lock file by running a stack command that accesses dependencies
   stack ls dependencies --test > /dev/null
-  # freeze cabal dependencies
-  $(dirname $0)/freeze-cabal-to-stack-resolver.sh
-  # since we use cabal.project as source of truth for nix builds, we need to provide
-  # a valid --sha256 hash for the haskell-backend library
-  # the easiest way to do this is to simply run nix, let it fail and substitute the
-  # correct sha back into cabal.project
-  SHA=$(nix build 2>&1 | grep -oP 'got:\s+\K.+') || true
-  sed -i -r "/haskell-backend.git/{n;s/tag:.*$/tag: $REV/;n;s|--sha256:.*$|--sha256: $SHA|}" cabal.project
```

Comment:
We don't use cabal for dependency management at all any more, since we either use non-nix stack or nix, which actually relies on stack via stacklock2nix and provides a custom version of cabal with all the correct dependencies from stack.